src/workerd/api/node/crypto_random.c++

```diff
@@ -0,0 +1,85 @@
+#include "crypto.h"
+#include "crypto_util.h"
+#include <v8.h>
+#include <openssl/bn.h>
+#include <openssl/rand.h>
+#include <workerd/jsg/jsg.h>
+#include <workerd/api/crypto-impl.h>
+
+namespace workerd::api::node {
+  kj::Array<kj::byte> CryptoImpl::randomPrime(uint32_t size, bool safe,
+      jsg::Optional<kj::Array<kj::byte>> add_buf, jsg::Optional<kj::Array<kj::byte>> rem_buf) {
+  BIGNUM* add = nullptr;
+
+  // Use mapping to have kj::Own work with optional buffer, perhaps there's a better way to format this
```

Comment:
Since you're doing this conversion to a `kj::Own<BIGNUM>` twice, it might make sense to encapsulate it into a const auto lambda to reduce code duplication

```
const auto maybeOwnBignum = [&](jsg::Optional<kj::Array<kj::byte>> maybeBignum) {
  return maybeBignum.map([](kj::Array<kj::byte>& a) {
    return OSSLCALL_OWN(BIGNUM, BN_bin2bn(a.begin(), a.size(), nullptr),
        RangeError, "Error importing add parameter", internalDescribeOpensslErrors());
    });
};
```

Also, these should likely use `TypeError` or `RangeError` instead of a `DOMException`. The Node.js crypto API generally does not use `DOMException` at all.

Reply:
Thank you, that's a lot cleaner